### PR TITLE
Hotfix/Cross device send link options affecting host app/site page routing

### PR DESCRIPTION
**Pull request by kopijunkie**
# Problem
On some client integrations when user is on the Cross Device "Send links" screen clicking on the "Get link via SMS" and/or "Copy link" view option links cause host app/site to go back in the browser history or causes the host app/site to refresh hence potentially preventing users from completing the flow if the Web SDK is not initialised to be in modal mode.

# Solution
- Call for `preventDefaultOnClick` util function was being done incorrectly
- Added checks in UI tests for Cross Device "Send links" screen UX to verify that browser URL does not get modified on clicking on the "Get link via SMS" and "Copy link" view options

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [ ] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1410-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1410-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1410-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1410-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1410-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1410-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1410-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1410-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1410-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1410-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/crossDevice/CrossDeviceLink/index.js [hunk 1]
```diff
@@ -350,9 +350,9 @@ class CrossDeviceLinkUI extends Component {
                   style[view.className]
                 )}
                 ref={(node) => (this.viewOptionBtn = node)}
-                onClick={() =>
-                  preventDefaultOnClick(this.handleViewOptionSelect(view.id))
-                }
+                onClick={preventDefaultOnClick(() =>
```
Comment (ethan605) [inline src/components/crossDevice/CrossDeviceLink/index.js:353]: Out of scope: IMO we'll have troubles in the future with this helper when converting the related files into TypeScript. So far no other places (which consume this helper) were converted to TypeScript so the risks are still unknown.
Btw I prefer to make it explicit: just call `event.preventDefault()` in-place, we probably don't need this helper.

**Review by ethan605 — APPROVED** [review]
LGTM, just a minor concern of `preventDefaultOnClick`

File: src/components/crossDevice/CrossDeviceLink/index.js (hunk 1 above)
Comment (kopijunkie) [reply src/components/crossDevice/CrossDeviceLink/index.js:353]: Yeah, I agree. I find it's clearer when `event.preventDefault()` is called in place & we don't have to deal with passing the next/actual function at onClick event on to the helper as a callback correctly. Which is the mistake I made here. 🙈

**Review by stefaniacardenas — APPROVED** [review]

**Pull request merged**